clientconn.go

```diff
@@ -823,6 +835,17 @@ func (cc *ClientConn) handleServiceConfig(js string) error {
 	return nil
 }
 
+func (cc *ClientConn) resolveNow(o resolver.ResolveNowOption) {
+	cc.mu.Lock()
+	r := cc.resolverWrapper
+	if r == nil {
+		cc.mu.Unlock()
+		return
+	}
+	cc.mu.Unlock()
+	go r.resolveNow(o)
```

Comment:
Can this result in multiple simultaneous outgoing calls to the resolver's `ResolveNow` method?  Is that OK?

Reply:
It's up to the resolver on how to deal with parallel `ResolveNow()`. In the dns resolver implementation, a new `ResolveNow()` is ignored if the previous one hasn't been processed.

From ClientConn's point, if multiple `ResolveNow()` result in multiple `NewAddress()` calls, only the last update will be kept. And we also ignore this update if it's the same as the cache we have.

Reply:
Maybe we should add a comment to ResolveNow() to let users know that it could be called multiple times concurrently.

Reply:
Done